test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,230 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CleaningPolicy, CacheModeTrait
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools.dd import Dd
+from test_tools.fs_utils import create_random_test_file
+from test_tools.iostat import IOstat
+from test_utils.filesystem.file import File
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+
+bs = Size(512, Unit.KibiByte)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_clean_stop_cache(cache_mode):
+    """
+        title: Test of the ability to stop correctly in modes with lazy writes.
+        description: |
+          Test if OpenCAS stops cache in modes with lazy writes without data loss.
+        pass_criteria:
+          - Cache stopping works properly.
+          - Data on core device is correct after cache is stopped.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(256, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step("Add core to cache."):
+        core = cache.add_core(core_part)
+
+    with TestRun.step("Change cleaning policy to 'NOP'."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Write data to exported object."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        dd = Dd().output(core.system_path) \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+
+    with TestRun.step("Check IO stats."):
+        # 4kB is written when partition is created
+        core_stats = (IOstat.get_iostat(core_part).total_writes - Size(4, Unit.KibiByte)).value
```

Comment:
You should disable seq cut-off to make sure that statistics have proper values, because dd is runnning sequential I/O, which after gaining  set threshold will be run in pass-through